### DISCUSSION TITLE
Find session with no connected users from the command line

### DISCUSSION THREAD

**Discussion by soenkeliebau** [zellij-org/zellij#3774] [Q&A]
**Context** 
I have recently (well, its been months..) switched to zellij from tmux. I have my terminal (alacritty) configured to run zellij on startup, so it pretty much never comes up without zellij and when I close the terminal the zellij session is exited (detached would be the tmux term I guess - I'll use exited and detached synonymously for the rest of the post :) ).

For tmux I had a script that listed sessions, grepped if there are any detached sessions available and attached the first detached session if there was one - or created a new one if none were detached.
I had that script bound to alacritty, so whenever I opened a new terminal, it would either attach to an existing session that was currently detached / exited  or create a new session.

I'd like to recreate this behavior in zellij, in principle the same script could be used, but zellij thinks about sessions a bit differently.
I currently have two terminals open, and when I look at my current session manager, it looks like this:

![image](https://github.com/user-attachments/assets/d924df6a-cf49-4be3-b487-d8e05953192a)

So all sessions are "not exited", rather the check I'll want to do is "which ones have no connected users?" 

In the example above, I'd like to attach to `chatty-accordion'. All fine and dandy, but the output of `zellij list-sessions` sadly doesn't contain the connected users:

```
❯ zellij list-sessions
rusty-weasel [Created 2h 17m 32s ago]
sensible-zebra [Created 2h 3m 59s ago]
chatty-accordion [Created 2h 2m 57s ago]
circular-zebra [Created 2h 2m 9s ago]
remarkable-elephant [Created 1h 59m 50s ago]
judicious-rustacean [Created 1h 50m 3s ago]
implacable-rhinoceros [Created 1h 26m 37s ago]
erudite-lemon [Created 1h 25m 50s ago]
wise-echidna [Created 1h 1m 52s ago]
charming-pheasant [Created 2s ago] (current)
```

**Question**
Is there a way to include the number of connected users per session in the output of `list-sessions` from the command line? 

Ideally I'd like a json representation of current sessions that lends itself well to parsing, but I'd also be happy to work with a text representation..

**Comment by soenkeliebau** [comment]
I have looked through the code a bit, and the information for the session manager and the cli command seem to come from totally different places .. 
As far as I can tell the cli command purely looks at the cache folder with serialized sessions, while the information the session manager uses comes from subscribing to the Events api and keeping an up to date representation of the current sessions and their connected users in memory ..

This sparked the idea of maybe being able to use `zellij pipe ...` to send a message to the session-manager and getting that state back as json, which is not currently supported, but might be a somewhat trivial change ... 

I'll maybe play around with that tonight a bit .. unless someone tells me "stop being stupid and just use this obvious way of doing it" :)

**Comment by giyany** [comment]
Hello! Do you know about the zellij welcome screen?
https://zellij.dev/news/welcome-screen-pipes-filepicker/#welcome-screen

It is not as automated as your own script, which would definitely be a nice solution - and a really nice plugin. It seems that others are halfway there, for example: https://github.com/mostafaqanbaryan/zellij-switch
